### DISCUSSION TITLE
After updating from ArgoCD 1.4 to 2.0

### DISCUSSION THREAD

**Discussion by dlebee** [argoproj/argo-cd#6049] [Q&A]
Hello, I updated my argoCD from 1.4 to 2.0.

Everything seems to work fine, after needing to reconfigure some stuff.

My issue is when a PVC was created with Tiller in the past, and ArgoCD wants to create using Helm now.

![image](https://user-images.githubusercontent.com/36390643/115067763-5bbd3200-9ebf-11eb-8cb8-f3eb828f5ce5.png)

It causes a sync error because some of the values are immutable.

Is there any solution to fix this manner? except having to re-deploy a fresh new MariaDB & Redis?

Here are some screenshots to show the issue:

![image](https://user-images.githubusercontent.com/36390643/115067468-f9fcc800-9ebe-11eb-9fb1-c6e6f26dfd65.png)

Thank you,
David.

**Comment by jannfis** [comment]
Hey, what you see is rooted in the fact that Argo CD v2.0 now uses Helm 3 as the default tool to render Helm charts, and this will generate values for the `heritage` label. This is covered [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/1.8-2.0/#helm-v3-is-now-the-default-when-rendering-charts). You can explicitly set back to Helm v2, which will use the same value in the `heritage` label as before.

Alternatively, you might also have success with new feature that uses `replace` instead of `apply`: https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#replace-resource-instead-of-applying-changes
